Jellyfin.Api/Controllers/LibraryController.cs

```diff
@@ -373,16 +381,25 @@ public async Task<ActionResult> DeleteItems([FromQuery, ModelBinder(typeof(Comma
                 var item = _libraryManager.GetItemById(i);
                 var auth = await _authContext.GetAuthorizationInfo(Request).ConfigureAwait(false);
                 var user = auth.User;
+                var authToken = auth.IsAuthenticated;
 
-                if (!item.CanDelete(user))
+                if (user != null)
                 {
-                    if (ids.Length > 1)
+                    if (!item.CanDelete(user))
                     {
-                        return Unauthorized("Unauthorized access");
+                        if (ids.Length > 1)
+                        {
+                            return Unauthorized("Unauthorized access");
+                        }
                     }
 
                     continue;
                 }
+                else if (!authToken)
```

Comment:
What other checks could be used? The user check already existed, and would fail if no user was returned, like when running the command from terminal. The authToken is just an additional check in case the user check fails, then check to see if the token is valid before returning an authorization failure.

Reply:
Ah I see your point. I misread the original comment. I think you’re right that a valid token trumps a user. Though in this scenario, as far as I can tell, it’s one or the other. That is you either have an api token or a user returned, but not both. From what I could see you’ll never get a user returned during an api call. Though maybe I’m wrong and someone can provide further evidence.